gapis/api/gles/state_builder.go

```diff
@@ -276,9 +276,14 @@ func (sb *stateBuilder) contextObject(ctx context.Context, handle EGLContext, c
 		}
 	}
 	if objs := c.Objects().Programs(); sb.once(objs) {
+		// Get the largest used shader ID.
+		maxID := uint32(0)
+		if l := c.Objects().Shaders().Len(); l > 0 {
+			maxID = uint32(c.Objects().Shaders().Keys()[l-1])
```

Comment:
Some assumptions here about sequential keys which I'm not sure is guaranteed. Would iterating over all shaders to take the maximum id field really too slow?

Reply:
Done. I knew keys are sorted, but thought `.Keys()` was constant time...